From bfb9b08157155124f3ffed840a394d460927e2fa Mon Sep 17 00:00:00 2001
From: Hien Dang <hien.dang.eb@rvc.renesas.com>
Date: Wed, 16 Mar 2016 10:34:49 +0700
Subject: [PATCH 1288/2066] thermal: rcar_gen3_thermal: Modify initialization
 flow

Modify initialization flow following the latest HWM.

Signed-off-by: Hien Dang <hien.dang.eb@rvc.renesas.com>
[zou:Original patch taken from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git
 v4.9/rcar-3.5.1]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/thermal/rcar_gen3_thermal.c |   13 ++++++++-----
 1 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/thermal/rcar_gen3_thermal.c b/drivers/thermal/rcar_gen3_thermal.c
index ef97a41..7cb7466 100644
--- a/drivers/thermal/rcar_gen3_thermal.c
+++ b/drivers/thermal/rcar_gen3_thermal.c
@@ -410,18 +410,21 @@ static int rcar_gen3_thermal_init(struct rcar_thermal_priv *priv)
 
 	spin_lock_irqsave(&priv->lock, flags);
 
+	rcar_thermal_write(priv, REG_GEN3_CTSR,  THBGR);
 	rcar_thermal_write(priv, REG_GEN3_CTSR,  0x0);
 
-	rcar_thermal_write(priv, REG_GEN3_CTSR,
-			PONM | AOUT | THBGR | VMEN);
-	udelay(100);
-	rcar_thermal_write(priv, REG_GEN3_CTSR,
-			PONM | AOUT | THBGR | VMEN | VMST | THSST);
 	udelay(1000);
 
+	rcar_thermal_write(priv, REG_GEN3_CTSR, PONM);
 	rcar_thermal_write(priv, REG_GEN3_IRQCTL, 0x3F);
 	rcar_thermal_write(priv, REG_GEN3_IRQEN, TEMP_IRQ_SHIFT(priv->id) |
 						TEMPD_IRQ_SHIFT(priv->id));
+	rcar_thermal_write(priv, REG_GEN3_CTSR,
+			PONM | AOUT | THBGR | VMEN);
+	udelay(100);
+
+	rcar_thermal_write(priv, REG_GEN3_CTSR,
+			PONM | AOUT | THBGR | VMEN | VMST | THSST);
 
 	spin_unlock_irqrestore(&priv->lock, flags);
 
-- 
1.7.5.4

